### PR TITLE
Adopt java.util.HashMap.newHashMap(numMappings)

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Reference - https://bugs.openjdk.java.net/browse/JDK-8186958
_JDK contains number of usages of HashMap, which are created with non-optimal initial capacity.
It seems that people normally assume that the constructor HashMap(int) requires the expected size as the argument.
In fact, the argument is the initial capacity, which should be bigger than the expected size. As a result, unnecessary reallocations can occur while the maps are filled._

Depends on https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/426 w/ https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/426/commits/dca73daef2bdcee73cd491532337e0c93929e03b

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by JasonFengJ9** [comment]
Updated following as well:
```
ComponentBuilder.java
ManagementUtils.java
ExtendedGarbageCollectorMXBeanImpl.java
```

**Comment by JasonFengJ9** [comment]
Fixed https://github.com/eclipse-openj9/openj9/blob/master/jcl/src/java.base/share/classes/java/lang/Class.java#L3689.
Pls have another look.

File: jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ManagementUtils.java [hunk 1]
```diff
@@ -892,7 +897,12 @@ private Component(String objectNamePattern, Collection<? extends T> beans) {
 			checkNames(beans, pattern);
 			/*[ENDIF]*/
 
-			this.beansByName = new HashMap<>(beans.size());
+			/*[IF JAVA_SPEC_VERSION >= 19]
+			this.beansByName = HashMap.newHashMap(beans.size());
+			/*[ELSE] JAVA_SPEC_VERSION >= 19 */
```
Comment (keithc-ca) [inline jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ManagementUtils.java:902]: This is not used: see line 604: `/*[IF !Sidecar19-SE]*/`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ManagementUtils.java (hunk 1 above)
Comment (JasonFengJ9) [reply jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ManagementUtils.java]: Removed Java 19 path.

File: jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ManagementUtils.java [hunk 2]
```diff
@@ -596,7 +601,7 @@ public static ObjectName createObjectName(String domain, String name) {
 	 */
 	public static final String BUFFERPOOL_MXBEAN_DOMAIN_TYPE = "java.nio:type=BufferPool"; //$NON-NLS-1$
 
-	/*[IF !Sidecar19-SE]*/
+/*[IF JAVA_SPEC_VERSION == 8]*/
```
Comment (keithc-ca) [inline jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ManagementUtils.java:604]: The indentation of this line and line 1032 was correct: please don't change that.

Comment (JasonFengJ9) [reply jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ManagementUtils.java]: Restored the tab.

**Comment by keithc-ca** [comment]
jenkins test sanity osx jdk11,jdknext

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**